lib/Controller/RoomController.php

```diff
@@ -1484,33 +1484,16 @@ public function setPassword(string $password): DataResponse {
 	 * 409: Session already exists
 	 */
 	#[PublicPage]
-	#[BruteForceProtection(action: 'talkFederationAccess')]
 	#[BruteForceProtection(action: 'talkRoomPassword')]
 	#[BruteForceProtection(action: 'talkRoomToken')]
 	public function joinRoom(string $token, string $password = '', bool $force = true): DataResponse {
 		$sessionId = $this->session->getSessionForRoom($token);
-		$isTalkFederation = $this->federationAuthenticator->isFederationRequest();
 		try {
 			// The participant is just joining, so enforce to not load any session
-			if (!$isTalkFederation) {
-				$action = 'talkRoomToken';
-				$room = $this->manager->getRoomForUserByToken($token, $this->userId, null);
-			} else {
-				$action = 'talkFederationAccess';
```

Comment:
But wouldn't [join requests from a remote Talk 19](https://github.com/nextcloud/spreed/blob/v19.0.0/lib/Federation/Proxy/TalkV1/Controller/RoomController.php#L93) go to [`joinFederatedRoom`](https://github.com/nextcloud/spreed/blob/main/appinfo/routes/routesRoomController.php#L68-L69) instead of `joinRoom`? In that case the request would not provide a session id, but in that case [Talk 20 should behave just like before](https://github.com/nextcloud/spreed/pull/12604/commits/a906acf79cd08c066609c211b1b0c8b2eac4545d#diff-d583983624ef8769460d4aa59e64b75ee54b6955e01e620e3eacedb7ba208786R1625).

Reply:
Hmm so this was never used then?

Reply:
It was added with https://github.com/nextcloud/spreed/pull/7514, but apparently proxied joins started to be used with https://github.com/nextcloud/spreed/pull/11698 :shrug:

Reply:
Ah, so from the old approach. Okay to remove then